frontend/src/console/components/CustomPages/CustomPages.tsx

```diff
@@ -0,0 +1,218 @@
+import { StaticContentOverrides } from 'ocim-client';
+import { Row, Col, Form } from 'react-bootstrap';
+import { Prompt } from 'react-router';
+import { Editor } from '@tinymce/tinymce-react';
+import messages from './displayMessages';
+
+interface State {
+  [key: string]: any;
+  pageContent: any;
+}
+interface ActionProps {
+  updateStaticContentOverridesFieldValue: Function;
+}
+interface StateProps extends InstancesModel {}
+interface Props extends StateProps, ActionProps {
+  // Passing URL as parameter to custom page
+  match: {
+    params: {
+      pageName: string;
+    };
+  };
+}
+
+export class CustomPagesComponent extends React.PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      pageContent: this.getPageContentFromState()
+    };
+  }
+
+  componentDidUpdate = (prevProps: Props) => {
+    const previousPageName = prevProps.match.params.pageName;
+    const currentPageName = this.props.match.params.pageName;
+
+    // If the page changed, clear edit field and render the correct
+    // override content.
+    if (previousPageName !== currentPageName) {
+      // Since we know for sure this won't generate a infinite loop, we can
+      // ignore the setState check inside componentDidUpdate for the next line
+      // eslint-disable-next-line
+      this.setState({
+        pageContent: this.getPageContentFromState()
+      });
+    }
+
+    // If page is receiving new instance data check and update internal state
+    // This prevents the internal state to be "" (empty) if data isn't available
+    // when the component is created but is available later.
+    if (
+      prevProps.activeInstance.data == null &&
+      this.props.activeInstance.data
+    ) {
+      // eslint-disable-next-line
+      this.setState({
+        pageContent: this.getPageContentFromState()
+      });
+    }
+  };
+
+  getApiPageName = () => {
+    const pageName = capitalizeFirstLetter(this.props.match.params.pageName);
+    return `staticTemplate${pageName}Content` as keyof StaticContentOverrides;
+  };
+
+  getPageContentFromState = () => {
+    let content = '';
+    const instance = this.props.activeInstance;
+
+    if (
+      instance &&
+      instance.data &&
+      instance.data.draftStaticContentOverrides
+    ) {
+      const statePageContent =
+        instance.data.draftStaticContentOverrides[this.getApiPageName()];
+      if (typeof statePageContent === 'string') {
+        content = statePageContent;
+      }
+    }
+
+    return content;
+  };
+
+  hasContentChanged = () => {
+    return this.state.pageContent !== this.getPageContentFromState();
+  };
+
+  onChangeContent = (newContent: any) => {
+    this.setState({
+      pageContent: newContent
+    });
+  };
+
+  getLMSLinkForPage = () => {
+    const instance = this.props.activeInstance;
+    const { pageName } = this.props.match.params;
+
+    if (instance && instance.data && instance.data.lmsUrl) {
```

Comment:
It's `undefined` for me, despite the fact that I'm having `lms_url` returned by `/api/v2/instances/openedx_config/` endpoint.

Reply:
@Agrendalath You have to rebuild the API client, otherwise the `lms_url` field won't be translated to `lmsUrl`.

Reply:
Ah, forgot about this step. Works great now!